### PR TITLE
View additions and tweaks

### PR DESCRIPTION
**Pull request by chrismccord**

### CODE REVIEW

File: docs/views.md [hunk 1]
```diff
@@ -50,7 +48,24 @@ When we reload the Welcome to Phoenix page, we should see our new title.
...
+  end
+end
+```
+
+Now if you fire up the server with `mix phx.server` and visit `http://locahost:4000`, you should see the following text below your layout header instead of the main template page:
+```
+rendering with asigns [:conn, :view_module, :view_template]
+```
+
+Pretty neat, right? At compile-time, phoenix precompiles all `*.html.eex` templates and turns them into `render/2` function clauses on their respective view modules. At runtime, all templates are already loaded in memory. There's no disk reads, complex file caching, or template engine computation involved. This is also why we were able to define functions like `title/0` in our `LayoutView` and they were immediately available inside the layout's `app.html.eex` – the call to `title/0` was just a local function call!
```
Comment (Gazler) [inline docs/views.md]: s/phoenix/Phoenix?

File: docs/views.md [hunk 2]
```diff
@@ -108,14 +123,11 @@ This is the message: <%= message() %>
...
-
-The return value is a tuple beginning with the atom `:safe` and the resultant string of the interpolated template.
-
-"Safe" here means that Phoenix has escaped the contents of our rendered template. Phoenix defines its own `Phoenix.HTML.Safe` protocol with implementations for atoms, bitstrings, lists, integers, floats, and tuples to handle this escaping for us as our templates are rendered into strings.
+As we can see, we're calling `render/3` with the individual view responsible for our test template, the name of our test template, and an empty map representing any data we might have wanted to pass in. The return value is a tuple beginning with the atom `:safe` and the resultant string of the interpolated template. "Safe" here means that Phoenix has escaped the contents of our rendered template. Phoenix defines its own `Phoenix.HTML.Safe` protocol with implementations for atoms, bitstrings, lists, integers, floats, and tuples to handle this escaping for us as our templates are rendered into strings.
```
Comment (Gazler) [inline docs/views.md]: Is it worth mentioning that the tuple returns `:safe` and an io list instead of a string?

File: docs/views.md [hunk 3]
```diff
@@ -264,33 +268,19 @@ Phoenix generates an `ErrorView` for us, but it doesn't give us a `lib/hello_pho
...
-
-We can also use the `assigns` map passed into any `render/2` clause in the `ErrorView`, instead of discarding it, in order to display more information in our templates.
+When we go back to [http://localhost:4000/such/a/wrong/path](http://localhost:4000/such/a/wrong/path), we should see a much nicer error page. It is worth noting that we did not render our `404.html.eex` template through our application layout, even though we want our error page to have the look and feel of the rest of our site. The main reason is that it's easy to run into edge case issues while handling errors globally. If we want to minimize duplication between our application layout and our `404.html.eex` template, we can implement shared templates for our header and footer. Please see the [Template Guide](http://www.phoenixframework.org/docs/templates#section-shared-templates-across-views) for more information. Of course, we can do these same steps with the `def render("500.html", _assigns) do` clause in our `ErrorView` as well. We can also use the `assigns` map passed into any `render/2` clause in the `ErrorView`, instead of discarding it, in order to display more information in our templates.
```
Comment (Gazler) [inline docs/views.md]: We should change the template guide link to an internal reference.

**Comment by chrismccord** [comment]
@Gazler I agree on all changes. Can you make them against this branch and wrap things up? I'm on context guide today

**Comment by Gazler** [comment]
Leave it with me and I'll try to get it done later today.

**Comment by Gazler** [comment]
@chrismccord Done. Squash and merge if you are happy with it.

**Comment by chrismccord** [comment]
😍😍😍😍

**Pull request merged**